### PR TITLE
Improve logging and visibility

### PR DESCRIPTION
**Pull request by christophermaier**
Dump even more information into the build logs to expose precisely what is going on at every step.

Read individual commit messages for further details.

### CODE REVIEW

File: lib/log.sh [hunk 1]
```diff
@@ -1,9 +1,46 @@
...
+    echo "${CSI}${BOLD};${color}${END}${*}${RESET}"
+}
+
+function bright_white() {
+    _bold_color "${WHITE}" "${@}"
+}
+
+# Logging
+########################################################################
+# NOTE: All logs get sent to standard error
```
Comment (twunderlich-grapl) [inline lib/log.sh:32]: Why are we sending all logs to stderr instead of stdout?

**Review by twunderlich-grapl — APPROVED** [review]

File: lib/log.sh (hunk 1 above)
Comment (christophermaier) [reply lib/log.sh:32]: I want to be able to add log messages wherever I want to, without having to worry about screwing up the "real" output of a function.

File: lib/rc.sh [hunk 2]
```diff
@@ -112,7 +114,7 @@ fetch_rc_config() {
     mv "${_temp_file}" "${_temp_file}.yaml"
     local _temp_file="${_temp_file}.yaml"
 
-    git show "origin/rc:${_stack_file_path}" > "${_temp_file}"
+    log_and_run git show "origin/rc:${_stack_file_path}" > "${_temp_file}"
```
Comment (wimax-grapl) [inline lib/rc.sh:117]: (i'm concerned that the output here would include the log_and_run output)

**Review by wimax-grapl — COMMENTED** [review]
(while it wouldn't be as sexy, `set -o xtrace` would complete a lot of the same things)

**Pull request merged**